packages/api/src/client/client.ts

```diff
@@ -324,27 +327,197 @@ export class MessariClient extends MessariClientBase {
 
       // Check if the response is JSON or text based on Content-Type header
       const contentType = response.headers.get("Content-Type");
-      let responseData: { data: T };
 
       if (contentType?.toLowerCase().includes("application/json")) {
-        responseData = await response.json();
-      } else {
-        responseData = { data: await response.text() } as { data: T };
+        const jsonResponse = await response.json();
+        // If response has data field and no error, unwrap it, otherwise use the whole response
+        const data = jsonResponse.data && !jsonResponse.error ? jsonResponse.data : jsonResponse;
```

Comment:
Instead of parsing it into a data object and then unwrapping it later we just check if a data object already exists and there's no error and if so we return that raw object or if there is no data but also no error then we just parse the JSON response as is.